### PR TITLE
[Windows]: Allow LoadLibraryExW

### LINKED ISSUE #65: Allow altering search path - add LoadLibraryExW on Windows
We've got a scenario where we want to alter the search path of LoadLibrary, which is done via the flags passed to [LoadLibraryExW](https://docs.microsoft.com/en-us/windows/win32/api/libloaderapi/nf-libloaderapi-loadlibraryexw).

For example, we would like to avoid searching the user's PATH for the library, and only search the application directory - which is the flag [LOAD_LIBRARY_SEARCH_APPLICATION_DIR](https://docs.rs/winapi/latest/winapi/um/libloaderapi/index.html).

So this means adding another `new` function to the windows-specific module, which calls `LoadLibraryExW`, and allows passing flags. (Note they may be combined, so we can use the [bitflags](https://docs.rs/bitflags/1.2.1/bitflags/) crate for that)

Relevant issues:
https://github.com/rust-windowing/glutin/issues/1292
https://github.com/feenkcom/gtoolkit/issues/973

### PR DESCRIPTION
**Pull request by NightRa**
Fixes #65

### CODE REVIEW

**Comment by nagisa** [comment]
LGTM, will merge and release a version with this within an hour.

File: src/error.rs [hunk 1]
```diff
@@ -32,9 +32,9 @@ pub enum Error {
     /// The `dlclose` call failed and system did not report an error.
     DlCloseUnknown,
     /// The `LoadLibraryW` call failed.
```
Comment (nagisa) [inline src/error.rs:34]: These documentation may need to be adjusted as well.

File: src/error.rs [hunk 2]
```diff
@@ -32,9 +32,9 @@ pub enum Error {
...
-    LoadLibraryW { source: WindowsError },
+    LoadLibraryExW { source: WindowsError },
```
Comment (nagisa) [inline src/error.rs:35]: Consider not removing the old variants quite yet, as it makes what could be a non-breaking change a breaking one.

**Comment by NightRa** [comment]
@nagisa Want to merge & do the little fix-ups?

**Comment by NightRa** [comment]
@nagisa 
One path to backwards compat would be leaving the old variant name, and renaming it on the next breaking change - so there needs to be a visible TODO for that somewhere.

**Comment by nagisa** [comment]
Yeah I can do the fixups myself, just noted them so that I don’t miss them later.

**Comment by nagisa** [comment]
Released 0.6.1 with this PR included.

**Comment by nagisa** [comment]
[Changelog](https://docs.rs/libloading/0.6.1/x86_64-pc-windows-msvc/libloading/changelog/r0_6_1/index.html).

**Pull request merged**